src/conv/k-g/k_parser.cpp

```diff
@@ -254,6 +362,25 @@ bool parse_k
 			    else
 				std::cout << "Unexpected command " << tokens[0] << " in k-file " << fileName << "\n";
 			}
+			else if (command[0] == "KEYWORD") {
+			    if (tokens.size() == 2) {

```

Comment:
I looks like there could be many keys in any order?
Therefore I would recommend to loop through them with a for-loop, split the keys at the "=" character, and decide then about the format. The last one has it?

The split could be done in an own function: `std::pair<std::string, std::string> split_key(const char* key)`.

BTW, maybe some of the `const char*` could be replacved by `const std::string&`? :thinking:

Reply:
yes there is usually mutiple keys (Formats), but I don't know if they come in any order. I will check it out.

Reply:
it's actually more complicated, they *Keyword options are 1- Number i.e (*KEYWORD 120m) this requests for 120 million words of memeory. 2- memory2=j, i.e ( *KEYWORD 120m memory2=j) this defines the memory allocation in words for the first core and j defines the memory allocation in words for each of the remaining cores. 3- NCPU i.e (*KEYWORD NCPU=n) it defines the number Shared Memory Parallel (SMP) threads of "n" to be used for each processor during the analyhsis. all these doesn't affect the format. for the format you have the standard where is nothing written after the *KEYWORD. For Long format you find LONG=S or LONG=Y or LONG=S. For I10 there is written I10 then LONG=S i.e (*KEYWORD I10 LONG=S). I don't konw if splitting the keys with the function will be any more different. but I think it will be nicer. I will prepare it